src/main/java/dev/openfeature/sdk/ProviderState.java

```diff
@@ -4,5 +4,17 @@
  * Indicates the state of the provider.
  */
 public enum ProviderState {
-    READY, NOT_READY, ERROR;
+    READY, NOT_READY, ERROR, STALE;
+
+    /**
+     * Returns true if the passed ProviderEvent maps to this ProviderState.
+     * 
+     * @param event event to compare
+     * @return boolean if matches.
+     */
+    boolean matchesEvent(ProviderEvent event) {
```

Comment:
```suggestion
    boolean isProviderEvent(ProviderEvent event) {
```

Note - you need to change the usage along with the renaming

Also,  we are missing `NOT_READY` state check

Reply:
There's no "not ready" event to match.

Reply:
ahh yes, you are right. I was little confused here - ProviderState vs ProviderEvent

I think this naming is fine :)